ivadomed/adaptative.py

```diff
@@ -517,18 +515,19 @@ def __getitem__(self, index):
                                 }
         """
         line = self.dataframe.loc[index]
-
-        input_tensors = [line[ct] for ct in self.cst_lst]
+        # For HeMIS strategy. Otherwise the values of the matrix dont change anything.
+        missing_modalities = self.cst_matrix[index]
 
         input_metadata = []
         input_tensors = []
 
         # Inputs
-        for ct in self.cst_lst:
+        for i, ct in enumerate(self.cst_lst):
             if self.status[ct]:
-                input_tensor = line[ct]
+                input_tensor = line[ct] * missing_modalities[i]
             else:
-                input_tensor = self.hdf5_file[line[ct]][line['Slices']]
+                input_tensor = self.hdf5_file[line[ct]][line['Slices']] * missing_modalities[i]
```

Comment:
I've added the **"cst_matrix"** (stands for contrast matrix) in the **Dataset variables.** The latter is basically a matrix full of "ones" except in the **HeMIS strategy** where some _"ones"_ are replaced by _"zeros"_ with a certain probability. That probability increase with time, which means that more and more modalities are missing over time. 

By using the method [**update(p)**](https://github.com/neuropoly/ivado-medical-imaging/blob/ab/adaptative_learning/ivadomed/adaptative.py#L576), we can change which modalities are missing. 

If the modality of a slice is missing, the slice is replaced by an array of "zeros". Besides the index of that modality is given to **HeMIS-Unet** to make sure that we do not take it into account when computing both **Mean** and **Var**. (see **HeMIS-Unet** architecture for more details)

Reply:
! (--> this is a sign of appreciation / excitation)

And you are filling your encoders with zeros when missing modalities to ensure a constant batch_size, is it correct?

Reply:
Yes exactly!